### PR TITLE
chore: fix melange in the main default devShell

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -49,7 +49,18 @@
       };
       pkgs = nixpkgs.legacyPackages.${system}.appendOverlays [
         (self: super: {
-          ocamlPackages = self.ocaml-ng.ocamlPackages_4_14;
+          ocamlPackages = self.ocaml-ng.ocamlPackages_4_14.overrideScope' (oself: osuper: {
```
Comment (rgrinberg) [inline flake.nix:52]: Shouldn't this overlay come from the melange flake somehow? It seems weird that dune has to know about this

Comment (anmonteiro) [reply flake.nix:52]: Ah, no, this is because we're using the melange overlay (to avoid compiling an extra ocaml 4.14 from my overlays).

Comment (rgrinberg) [reply flake.nix:52]: It's not possible for the melange overlay to include this?

Comment (rgrinberg) [reply flake.nix:52]: Sorry, I don't know much about nix.

Comment (anmonteiro) [reply flake.nix:52]: I think the alternative is:

- switch back to `melange.outputs.packages.${system}.default`
  - 👍  gives us the correct dune version, tracked in the melange flake
  - 👎 requires compiling OCaml+flambda from the ocaml overlays

Comment (anmonteiro) [reply flake.nix:52]: your alternative should also work: override dune_3 in the melange overlay. I can switch to that approach if you prefer.

**Review by Alizter — APPROVED** [review]
Works for me

**Pull request merged**